src/metabase/driver/postgres.clj

```diff
@@ -982,19 +982,48 @@
   [driver]
   (= driver :postgres))
 
-(defmethod sql-jdbc.sync/alter-columns-sql :postgres
-  [driver table-name column-definitions]
+(defn- alter-column-using-hsql-expr
+  "Returns a honey expr to convert the column from the old type to the new when a USING expr is necessary (nil if not).
+
+  The output can be used in the context of ALTER COLUMN when the :old-type key is provided
+  as part of the new column-definition.
+
+  Unfortunately out of the box not all trivial conversions are possible using CAST alone:
+  e.g. BOOLEAN to BIGINT, or BOOLEAN to FLOAT.
+  This function therefore provides more options for alter-columns-sql to produce DDL that can 'just work' without
+  extra input from the user."
+  [column new-type old-type]
+  (condp = [new-type old-type]
+    [[:bigint] [:boolean]]
```

Comment:
Cosmetic nit, but I think its more intuitive to order the pair like `[old new]`, as if they were key entries in a mapping of what was supported.

Reply:
Agree, good spot. https://github.com/metabase/metabase/pull/52607/commits/51c8a83a62761ab3c45937006a89b482f98ffa9e